service/role_service/role_service.go

```diff
@@ -0,0 +1,122 @@
+package role_service
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	db "example/employee/server/db/sqlc"
+	"fmt"
+	"log"
+)
+
+type RoleService struct {
+	Context context.Context
+	Store   db.Store
+}
+
+func NewRoleService(store db.Store, context context.Context) *RoleService {
+	return &RoleService{
+		Context: context,
+		Store:   store,
+	}
+}
+
+type Role string
+
+const (
+	Admin Role = "Admin"
+	Staff Role = "Staff"
+)
+
+var adminPriviledge = []db.Privilege{
+	db.PrivilegeCreateAndUpdateJobs,
+	db.PrivilegeCreateAndUpdateDepartments,
+	db.PrivilegeDeleteJobs,
+	db.PrivilegeDeleteDepartments,
+	db.PrivilegeCreateAndUpdateEmployees,
+	db.PrivilegeDelteEmployees,
+	db.PrivilegeReadAllEmployees,
+}
+
+type RolePrivilege struct {
+	pMap map[db.Privilege]bool
+}
+
+func GetAdminPrivilege() []byte {
+	pMap := map[db.Privilege]bool{}
+	for _, privilege := range adminPriviledge {
+		pMap[privilege] = true
```

Comment:
can apply goroutine here

Reply:
I think we need to use `concurrent-map` here if we use go-routine. Do you suggest to use go-routine + concurrent-map here?

Reply:
Let's just not increase the ticket scope here. We can always rise another ticket for parallel instruction. I will just merge it directly.